src/unix/pipe.c

```diff
@@ -302,3 +302,57 @@ uv_handle_type uv_pipe_pending_type(uv_pipe_t* handle) {
   else
     return uv__handle_type(handle->accepted_fd);
 }
+
+static int uv__pipe_chmod_validate(uv_pipe_t* handle, int mode) {
+  if (!handle || uv__stream_fd(handle) == -1)
+    return UV_EBADF;
+  if (mode != UV_READABLE &&
+      mode != UV_WRITABLE &&
+      mode != (UV_WRITABLE | UV_READABLE))
+    return UV_EINVAL;
+  return 0;
+}
+
+int uv_pipe_chmod(uv_pipe_t* handle, int mode) {
+  struct stat pipe_stat;
+  char* name_buffer;
+  size_t name_len;
+  int r;
+
+  r = uv__pipe_chmod_validate(handle, mode);
+  if (r != 0)
+    return r;
+
+  name_len = 0;
+  r = uv_pipe_getsockname(handle, NULL, &name_len);
+  if (r != UV_ENOBUFS)
+    return r;
+
+  name_buffer = uv__malloc(name_len + 1);
```

Comment:
`uv__malloc(name_len)` should be enough as `name_len` should include the terminating null character. ( I think the documentation in http://docs.libuv.org/en/v1.x/pipe.html#c.uv_pipe_getsockname is not correct)